packages/functional-tests/src/functional-tests.local.spec.ts

```diff
@@ -0,0 +1,169 @@
+import { isEmpty } from 'lodash';
+import { ConsoleLoggerClient, GlobalLogger } from 'logger';
+import {
+    OnDemandPageScanRunResultProvider,
+    PartitionKeyFactory,
+    RunState,
+    ScanResultResponse,
+    ScanRunResultResponse,
+} from 'service-library';
+import { A11yServiceClient, A11yServiceCredential } from 'web-api-client';
+
+import { TestEnvironment } from './common-types';
+import { functionalTestGroupTypes, TestGroupName } from './functional-test-group-types';
+import { TestRunner } from './runner/test-runner';
+import { TestContextData } from './test-group-data';
+import { FunctionalTestGroup } from './test-groups/functional-test-group';
+
+// tslint:disable: mocha-no-side-effect-code no-any no-unsafe-any mocha-unneeded-done strict-boolean-expressions
+
+describe('functional tests', () => {
+    // need to provide by user. DO NOT CHECK IN
+    const clientId = process.env.SP_CLIENT_ID || '';
+    const clientSecret = process.env.SP_PASSWORD || '';
+    const tenantId = process.env.SP_TENANT || '';
+    const apimName = process.env.APIM_Name || '';
+    const cosmosKey = process.env.COSMOS_DB_KEY || '';
+    const cosmosUrl = process.env.COSMOS_DB_URL || '';
+    // static values
+    const dbName = 'onDemandScanner';
+    const collectionName = 'scanRuns';
+    const releaseIdStub = 'release-id';
+    const runIdStub = 'run-id';
+
+    let consoleLoggerClient: ConsoleLoggerClient;
+    let logger: GlobalLogger;
+    let cred: A11yServiceCredential;
+    let a11yServiceClient: A11yServiceClient;
+    let testRunner: TestRunner;
+    let guidGenerator: GuidGenerator;
+    let testContextData: TestContextData;
+    let cosmosClientWrapper: CosmosClientWrapper;
+    let cosmosContainerClient: CosmosContainerClient;
+    let onDemandPageScanRunResultProvider: OnDemandPageScanRunResultProvider;
+
+    beforeAll(async () => {
+        if (isServiceCredProvided()) {
+            consoleLoggerClient = new ConsoleLoggerClient(new ServiceConfiguration(), console);
+            logger = new GlobalLogger([consoleLoggerClient], process);
+            cred = new A11yServiceCredential(clientId, clientSecret, clientId, `https://login.microsoftonline.com/${tenantId}`);
+            a11yServiceClient = new A11yServiceClient(cred, `https://apim-${apimName}.azure-api.net`);
+            testRunner = new TestRunner(logger);
+            guidGenerator = new GuidGenerator();
+            cosmosClientWrapper = new CosmosClientWrapper(async () => {
+                return new CosmosClient({ endpoint: cosmosUrl, auth: { masterKey: cosmosKey } });
+            }, logger);
+            cosmosContainerClient = new CosmosContainerClient(cosmosClientWrapper, dbName, collectionName, logger);
+            onDemandPageScanRunResultProvider = new OnDemandPageScanRunResultProvider(
+                cosmosContainerClient,
+                new PartitionKeyFactory(new HashGenerator(), guidGenerator),
+            );
```

Comment:
It seems  most of this code already implemented when we setup DI container. Can we reuse DI container implementation?

Reply:
Updated.

Reply:
Reused some DI implementation that is available. Need to refactor the 'getProcessLifeCycleContainer' in web-worker before we can fully reuse that. will do that in a separate PR.